var/spack/repos/builtin/packages/py-pymatgen/package.py

```diff
@@ -13,15 +13,27 @@ class PyPymatgen(PythonPackage):
     analysis code powering the Materials Project."""
 
     homepage = "http://www.pymatgen.org/"
-    pypi = "pymatgen/pymatgen-4.7.2.tar.gz"
+    pypi = "pymatgen/pymatgen-2023.3.10.tar.gz"
 
+    version("2023.3.10", sha256="20a5d1681e2951d2710d5358197b5b58992e2b0ec4217a5ba966cf28513c1283")
     version("2021.3.9", sha256="a6f22d69133a48b7801bfd5e6a2878b47b4b4b2ef1a377b87c6c573be14cbf16")
     version(
         "2020.12.31", sha256="5002490facd47c55d2dae42c35712e061c1f5d881180485c0543a899589856d6"
     )
 
     extends("python@3.7:", ignore="bin/tabulate")
 
+    # pymatgen does not support new recent python versions automatically.
+    # Update below when support for 3.12 arrives in newer versions:
+    conflicts("python@3.12:")
+    # See https://pymatgen.org/change_log.html for python version support:
+    conflicts("python@3.11:", when="@:2023.2.22")
+    # pymatgen@:2022.1.8 does not support python@3.10:
+    conflicts("python@3.10:", when="@:2022.1.08")
+    # This conflict only applies to older, not checksummed versions:
+    conflicts("python@3.9:", when="@:2020.10.9")
```

Comment:
It would be better to write these as `depends_on` so it's explicit what the range of versions supported are. For example, the latest version requires Python 3.8+ but that isn't encoded here. Also, in its current state, this doesn't actually work since it's missing the `^`.

Reply:
No, it works great: `spack.error.SpackError: py-pymatgen 'python@3.10:' conflicts with '@:2022.1.08'`

Reply:
Second question today on `conflicts` with `^`, the first one was in https://github.com/spack/spack/pull/36054#discussion_r1133564054 I always used `^` in conflicts, and I need to double check if there was a (non-tested) regression on that behavior.

Reply:
Needless to say, two unrelated questions in the same day raise this in the checklist priority :laughing: